test/generate-manifest.test.js

```diff
@@ -82,4 +82,30 @@ describe('Generate Complete Manifest', function () {
     const result = generateManifest(varsMock, manifestMock)
     return expect(result).to.eventually.become(validManifest)
   })
+
+  it('should produce an error if a container contains an invalid image', async function () {
+    const manifest = JSON.parse(JSON.stringify(manifestJson))
+    manifest['manifest']['containers'][0]['image'] = 'hello-world@1231984'
+    await fse.writeJson(manifestMock, manifest)
+    const result = generateManifest(varsMock, manifestMock)
+    return expect(result).to.be.rejected
+  })
+
+  it('should resolve image tags to proper digest', async function () {
```

Comment:
are these tests actually hitting the live docker hub? If so, we should try to mock out the network call instead

Reply:
@sharafian  normally I would agree but there could be some value in having this actually hit dockerhub acting like an integration test if it does change the api and stops working. 

Should we have some kind of docker integration test and a mocked version as well?

Reply:
Yeah, I suppose it's ok in that case. Probably not worth splitting into integration tests for a repo this small but maybe in the future

Reply:
Lets keep it but add a comment to the test or put it in a section integration tests by itself so its clear this this relies on external resources to pass.

Reply:
Looks good